### PR TITLE
Extend dialogs in debug mode

### PR DESCRIPTION
**Pull request by mueller-ma**
In debug mode url, username, password and stacktrace will be shown, when connections errors occure.

### CODE REVIEW

File: mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java [hunk 1]
```diff
@@ -115,36 +120,41 @@ public void onFailure(Call call, int statusCode, Headers headers, byte[] respons
...
+                message = getString(R.string.error_connection_failed);
             } else {
                 Log.e(TAG, error.getClass().toString());
-                showMessageToUser(error.getMessage(), Constants.MESSAGES.DIALOG, Constants.MESSAGES.LOGLEVEL.ALWAYS);
+                message = error.getMessage();
             }
+            showMessageToUser(message, DIALOG, NO_DEBUG);
+            message += "\nURL: " + openHABBaseUrl + "\nUsername: " + openHABUsername + "\nPassword: " + openHABPassword;
+            message += "\nStacktrace:\n" + Log.getStackTraceString(error);
+            showMessageToUser(message,DIALOG, DEBUG);
```
Comment (FlorianSW) [inline mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java]: missing whitespaces :)

Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java]: @FlorianSW Any other comments on that?

**Comment by mueller-ma** [comment]
I found the issue with the two messages. Should be gone now.
I tested it with 401 and I can see a stacktrace there.

File: mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java [hunk 2]
```diff
@@ -629,10 +629,9 @@ public void showMessageToUser(String message, int messageType, int logLevel) {
         String localUrl = mSettings.getString(Constants.PREFERENCE_URL, "");
 
         // if debug mode is enabled, show all messages, except those with logLevel 4
-        if(debugEnabled) {
-            if (logLevel == Constants.MESSAGES.LOGLEVEL.NO_DEBUG) {
-                return;
-            }
+        if((debugEnabled && logLevel == Constants.MESSAGES.LOGLEVEL.NO_DEBUG) ||
+                (!debugEnabled && logLevel == Constants.MESSAGES.LOGLEVEL.DEBUG)) {
+            return;
```
Comment (FlorianSW) [inline mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java]: btw.: Having a separate else branch here is not neccessary, as the return will already end this method :)

**Pull request merged**